react/features/toolbox/components/ToolboxV2.web.js

```diff
@@ -0,0 +1,408 @@
+                        iconName = { _videoMuted && _conference
+                            ? 'icon-camera-disabled toggled'
+                            : 'icon-camera' }
+                        onClick = { this._onToggleVideo }
+                        tooltip = { t('toolbar.videomute') } />
+                </div>
+                <div className = 'button-group-right'>
+                    <ToolbarButtonV2
+                        iconName = 'icon-add'
+                        onClick = { this._onShowInvite }
+                        tooltip = { _addPeopleAvailable || _dialOutAvailable
+                            ? t('addPeople.title')
+                            : t('addPeople.notAvailable') } />
+                    <InfoDialogButtonV2 />
+                    <OverflowMenuButton
+                        isOpen = { this.state.showOverflowMenu }
+                        onVisibilityChange = { this._onSetOverflowVisible }>
+                        <ul className = 'overflow-menu'>
+                            <OverflowMenuItem
+                                icon = 'icon-presentation'
+                                onClick = { this._onOpenSpeakerStats }
+                                text = { t('toolbar.speakerStats') } />
+                            <OverflowMenuItem
+                                icon = 'icon-feedback'
+                                onClick = { this._onOpenFeedback }
+                                text = { t('toolbar.feedback') } />
+                            <OverflowMenuItem
+                                icon = 'icon-link'
+                                onClick = { this._onOpenKeyboardShortcuts }
+                                text = { t('toolbar.shortcuts') } />
+                        </ul>
+                    </OverflowMenuButton>
+                </div>
+            </div>
+        );
+    }
+
+    _onHangup: () => void;
+
+    /**
+     * Dispatches an action to leave the current conference.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onHangup() {
+        sendAnalytics(createToolbarEvent('hangup'));
+        this.props.dispatch(disconnect(true));
+    }
+
+    _onMouseOut: () => void;
+
+    /**
+     * Dispatches an action signaling the toolbar is not being hovered.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onMouseOut() {
+        this.props.dispatch(setToolbarHovered(false));
+    }
+
+    _onMouseOver: () => void;
+
+    /**
+     * Dispatches an action signaling the toolbar is being hovered.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onMouseOver() {
+        this.props.dispatch(setToolbarHovered(true));
+    }
+
+    _onOpenFeedback: () => void;
+
+    /**
+     * Callback invoked to display {@code FeedbackDialog}.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onOpenFeedback() {
+        this.props.dispatch(openFeedbackDialog());
+
+        this._onSetOverflowVisible(false);
```

Comment:
This seems to be an action related to openFeedbackDialog. Couldn't it be easier to be handled though a redux action attached to openFeedbackDialog?

Reply:
Maybe the naming is wrong here. If you mean _onSetOverflowVisible, that's to hide the submenu that's attached to the toolbar. I'm not following what it would look like put that bit into a redux action attached to openFeedbackDialog.

Reply:
Okay, I think I see what you mean now and maybe where I was confused. _onSetOverflowVisible only affects localized state and not global app state (redux). I could move the overflow menu display into redux but I don't see a need just yet.

Reply:
It took me so long to figure out what you meant but I think I got the jive now. I don't know if it's exactly as you envisioned but now I check for modal openings on componentWillReceiveProps and close the overflow menu. Much less code floating around that way.